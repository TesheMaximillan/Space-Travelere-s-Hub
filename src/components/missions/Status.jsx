/* eslint-disable react/prop-types */
import React from 'react';
import styles from './Status.module.scss';

function Status({ status, handleStatus }) {
  const {
    btnMission, statusBg, btnLeave, statusActive,
  } = styles;

  if (status) {
    return (
      <>
        <td><span className={statusActive}>Active Member</span></td>
        <td>
          <button type="button" className={`${btnMission} ${btnLeave}`} onClick={handleStatus}>
            Leave Mission
          </button>
        </td>
      </>
    );
  }

  return (
    <>
      <td><span className={statusBg}>NOT A MEMBER</span></td>
      <td>
        <button type="button" className={btnMission} onClick={handleStatus}>
          Join Mission
        </button>
      </td>
    </>
  );
}

export default Status;
