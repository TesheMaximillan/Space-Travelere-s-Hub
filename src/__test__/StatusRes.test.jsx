/* eslint-disable no-whitespace-before-property */
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Status from '../components/rockets/StatusRes';

it('renders correctly', () => {
  const tree = renderer .create(
    <Provider store={store}>
      <Status />
      {' '}

    </Provider>,
  ) .toJSON();
  expect(tree).toMatchSnapshot();
});
