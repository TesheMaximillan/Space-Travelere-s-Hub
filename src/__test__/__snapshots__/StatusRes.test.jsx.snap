// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div>
    <span
      className="rocketsDescr"
    />
  </div>,
  <div>
    <button
      className="btnRocket"
      type="button"
    >
      Reservation
    </button>
  </div>,
  " ",
]
`;
